bevy_lint/docs/how-to/types.md

```diff
@@ -0,0 +1,74 @@
+# How to Work with Types
```

Comment:
Should we include a brief section on `TypeVisitor`?

Reply:
I haven't used `TypeVisitor`, so I wouldn't know what to write! Please feel free to open up a PR targeting this branch if you want to write your own section, though. :)

Reply:
Maybe in the future! Focusing on some other work before I come back to bevy_lint stuff, so don't consider this a blocker for this PR haha